bigquery/google/cloud/bigquery/job.py

```diff
@@ -1354,6 +1354,42 @@ def cache_hit(self):
         query_stats = self._query_statistics()
         return query_stats.get('cacheHit')
 
+    @property
+    def referenced_tables(self):
+        """Return referenced tables from job statistics, if present.
+
+        See:
+        https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#statistics.query.referencedTables
+
+        :rtype: list of dict
+        :returns: mappings describing the query plan, or an empty list
+                  if the query has not yet completed.
+        """
+        tables = []
+        client = self._require_client(None)
+        query_stats = self._query_statistics()
+        clients_by_project = {client.project: client}
+        datasets_by_project_name = {}
+
+        for table in query_stats.get('referencedTables', ()):
+
+            t_project = table['projectId']
+            t_client = clients_by_project.get(t_project)
+            if t_client is None:
+                t_client = client._clone(t_project)
```

Comment:
Looks like we should be able to set the project on the dataset without having to clone the client. https://googlecloudplatform.github.io/google-cloud-python/latest/bigquery/dataset.html